.travis.yml

```diff
@@ -1,195 +1,204 @@
+    - compiler: "ghc-7.8.3"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.8.3], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.8.2"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.8.2], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.8.1"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.8.1], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.6.3"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.6.3], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.6.2"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.6.2], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.6.1"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.6.1], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.4.2"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.4.2], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.4.1"
+      env: TEST=--disable-tests
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.4.1], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.2.2"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.2.2], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.2.1"
+      env: BATTERIES=NO TEST=--disable-tests
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.2.1], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.0.4"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.0.4], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.0.3"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.0.3], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.0.2"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.0.2], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.0.1"
+      env: BATTERIES=NO TEST=--disable-tests
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-7.0.1], sources: [hvr-ghc]}}
 
 before_install:
- - unset CC
- - export PATH=$HOME/.cabal/bin:/opt/ghc/$GHCVER/bin:/opt/cabal/$CABALVER/bin:$PATH
+  - HC=${CC}
+  - HCPKG=${HC/ghc/ghc-pkg}
+  - unset CC
+  - ROOTDIR=$(pwd)
+  - mkdir -p $HOME/.local/bin
+  # install typediff
+  - curl -L https://github.com/haskell-compat/base-compat/releases/download/typediff-0.1.3/typediff > $HOME/.local/bin/typediff
```

Comment:
The post/manual modifications are in a separate commit.

Reply:
What I'm requesting is some sort of text file (which lists all of the manual changes you had to make) that should be checked in alongside `.travis.yml`. Eventually, I'm going to need to regenerate and change `.travis.yml`, and it will be easy for that one particular commit which applied the hacks for the first time to get lost in the commit history. Having some prose which details all of these manual hacks will be extremely helpful for me (especially if we have to change any of these hacks later).